src/Data/Nat/Primality.agda

```diff
@@ -294,6 +294,17 @@ prime⇒rough (prime pr) = pr
 rough∧∣⇒prime : .{{NonTrivial p}} → p Rough n → p ∣ n → Prime p
 rough∧∣⇒prime r p∣n = prime (rough∧∣⇒rough r p∣n)
 
+-- If a number n is m-rough, and m * m > n, then n must be prime.
+rough∧square>⇒prime : .{{NonTrivial n}} → m Rough n → m * m > n → Prime n
+rough∧square>⇒prime rough m*m>n = prime ¬composite
+  where
+    ¬composite : ¬ Composite _
+    ¬composite (hasNonTrivialDivisor d<n d∣n) = contradiction (m∣n⇒n≡quotient*m d∣n)
```

Comment:
`composite` is a (very!) shallow pattern synonym for `hasNonTrivialDivisor` here, but I think it's helpful to have it as a constructor of `Composite`, just as `prime` is the constructor of `Prime`...